rules/expiring-todo-comments.js

```diff
@@ -0,0 +1,390 @@
+						node: null,
+						loc: comment.loc,
+						messageId,
+						data: {
+							package: dependency.name
+						}
+					});
+				}
+
+				continue;
+			}
+
+			const todoVersion = tryToCoerceVersion(dependency.version);
+			const targetPackageVersion = tryToCoerceVersion(targetPackageRawVersion);
+
+			if (!hasTargetPackage || !targetPackageVersion) {
+				// Can't compare ¯\_(ツ)_/¯
+				continue;
+			}
+
+			const compare = semverComparisonForOperator(dependency.condition);
+
+			if (compare(targetPackageVersion, todoVersion)) {
+				context.report({
+					node: null,
+					loc: comment.loc,
+					messageId: MESSAGE_ID_VERSION_MATCHES,
+					data: {
+						comparison: `${dependency.name} ${dependency.condition} ${
+							dependency.version
+						}`
+					}
+				});
+			}
+		}
+
+		const pkgEngines = pkg.engines || {};
+
+		for (const engine of engines) {
+			uses++;
+			const targetPackageRawEngineVersion = pkgEngines[engine.name];
+			const hasTargetEngine = Boolean(targetPackageRawEngineVersion);
+
+			if (!hasTargetEngine) {
+				continue;
+			}
+
+			const todoEngine = tryToCoerceVersion(engine.version);
+			const targetPackageEngineVersion = tryToCoerceVersion(
+				targetPackageRawEngineVersion
+			);
+
+			const compare = semverComparisonForOperator(engine.condition);
+
+			if (compare(targetPackageEngineVersion, todoEngine)) {
+				context.report({
+					node: null,
+					loc: comment.loc,
+					messageId: MESSAGE_ID_ENGINE_MATCHES,
+					data: {
+						comparison: `${engine.name} ${engine.condition} ${engine.version}`
+					}
+				});
+			}
+		}
+
+		return uses === 0;
+	}
+
+	return {
+		Program() {
+			rules.Program(); // eslint-disable-line new-cap
+		}
+	};
+};
+
+const schema = [
+	{
+		type: 'object',
+		properties: {
+			terms: {
+				type: 'array',
+				items: {
+					type: 'string'
+				}
+			},
+
+			ignoreDatesOnPR: {
+				type: 'boolean'
+			}
+		}
+	}
+];
+
+module.exports = {
+	create,
+	meta: {
+		type: 'suggestion',
+		docs: {
+			url: getDocsUrl(__filename)
+		},
+		messages: {
+			[MESSAGE_ID_AVOID_MULTIPLE_DATES]:
+				'Avoid using multiple expiration dates in TODO: {{ expirationDates }}',
+			[MESSAGE_ID_EXPIRED_TODO]:
+				'There is a TODO that is past due date: {{ expirationDate }}',
+			[MESSAGE_ID_REACHED_PACKAGE_VERSION]:
+				'There is a TODO that is past due package version: {{ comparison }}',
+			[MESSAGE_ID_AVOID_MULTIPLE_PACKAGE_VERSIONS]:
+				'Avoid using multiple package versions in TODO: {{ versions }}',
+			[MESSAGE_ID_HAVE_PACKAGE]:
+				'There is a TODO that is deprecated since you installed: {{ package }}',
+			[MESSAGE_ID_DONT_HAVE_PACKAGE]:
+				'There is a TODO that is deprecated since you uninstalled: {{ package }}',
+			[MESSAGE_ID_VERSION_MATCHES]:
+				'There is a TODO match for package version: {{ comparison }}',
+			[MESSAGE_ID_ENGINE_MATCHES]:
+				'There is a TODO match for engine version: {{ comparison }}'
+		},
+		schema
+	}
+};
+
+function parseTodoWithArgs(str, {terms}) {
```

Comment:
Don't use abbreviations.
```suggestion
function parseTodoWithArguments(string, {terms}) {
```

Reply:
Done